### ISSUE TITLE
resource aws_msk_cluster enhanced_monitoring does not allow setting to PER_TOPIC_PER_PARTITION

### ISSUE THREAD

**Issue by cemeyer2** [hashicorp/terraform-provider-aws#16866]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

* Terraform 0.12.28
* AWS provider 3.22.0

### Affected Resource(s)

* aws_msk_cluster

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_msk_cluster" "kafka_cluster" {
...
enhanced_monitoring = "PER_TOPIC_PER_PARTITION"
...
}
```

### Expected Behavior

The provider should provision MSK with the correct monitoring level specified.

### Actual Behavior

```
Error: expected enhanced_monitoring to be one of [DEFAULT PER_BROKER PER_TOPIC_PER_BROKER], got PER_TOPIC_PER_PARTITION
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

https://docs.aws.amazon.com/msk/latest/developerguide/metrics-details.html indicates `PER_TOPIC_PER_PARTITION` is a valid setting now, but https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/resource_aws_msk_cluster.go#L219-L223 seems to need to be updated to support this new value as one of the allowed parameters.

**Comment by anGie44** [comment]
Thank you for raising this issue @cemeyer2 , adding support should be a straightforward fix 👍 Just noting that we're moving in the direction of using the AWS Go SDK `_Values()` funcs to retrieve the possible list of values if available (for reference: https://github.com/hashicorp/terraform-provider-aws/issues/14601)

**Comment by gunadhya** [comment]
Hi @anGie44, I'd like to work on this issue. So I'm thinking the change would be something like this right?

**Before:** 
```javascript
"enhanced_monitoring": {
				Type:     schema.TypeString,
				Optional: true,
				Default:  kafka.EnhancedMonitoringDefault,
				ValidateFunc: validation.StringInSlice([]string{
					kafka.EnhancedMonitoringDefault,
					kafka.EnhancedMonitoringPerBroker,
					kafka.EnhancedMonitoringPerTopicPerBroker,
				}, true),
			}

```
**After:**
 ```javascript
"enhanced_monitoring": {
				Type:     schema.TypeString,
				Optional: true,
				Default:  kafka.EnhancedMonitoringDefault,
				ValidateFunc: validation.StringInSlice([]string{
					kafka.EnhancedMonitoring_Values()
				}, true),
			}

```

**Comment by anGie44** [comment]
`PER_TOPIC_PER_PARTITION` support has been merged and will release with `v3.23.0` of the Terraform AWS Provider, likely out this Thursday.
